### PR TITLE
Adding karma coverage to grunt

### PR DESCRIPTION
**Pull request by jloveland**
Adding karma coverage for displaying client side test coverage when running `grunt coverage`.
karma-coverage

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,6 +82,7 @@
     "grunt-ng-annotate": "^1.0.1",
     "grunt-node-inspector": "~0.3.0",
     "grunt-nodemon": "~0.4.0",
+    "grunt-protractor-coverage": "^0.2.15",
```
Comment (ilanbiala) [inline package.json]: ~

File: karma.conf.js [hunk 2]
```diff
@@ -30,7 +43,27 @@ module.exports = function (karmaConfig) {
 
     // Test results reporter to use
     // Possible values: 'dots', 'progress', 'junit', 'growl', 'coverage'
-    reporters: ['progress'],
+    reporters: karmaReporters,
```
Comment (ilanbiala) [inline karma.conf.js]: Why did you move this to a variable?

File: karma.conf.js [hunk 3]
```diff
@@ -30,7 +43,27 @@ module.exports = function (karmaConfig) {
...
+
+    // Configure the coverage reporter
+    coverageReporter: {
+      dir : 'coverage/',
+      reporters: [
```
Comment (ilanbiala) [inline karma.conf.js]: Don't we only need html/lcov?

**Comment by ilanbiala** [comment]
@jloveland good start. Can you also add Gulp support?

File: karma.conf.js (hunk 2 above)
Comment (jloveland) [inline karma.conf.js]: I added the variable so I can test for `testConfig.coverage === true` which is set by the grunt coverage task [here](https://github.com/meanjs/mean/pull/985/files#diff-595bf3fa2348192244b0319be33066b8R227)

File: karma.conf.js (hunk 3 above)
Comment (jloveland) [inline karma.conf.js]: we can add or remove options here. I think we can get away with html and text only. Text will output to the console for us. Does anyone want lcov?

**Comment by jloveland** [comment]
@ilanbiala thanks for the review. 

I would like to have the gulp karma coverage added in a different PR. Rationale is the following:
1. Mean.js doesn't fully support gulp yet
2. We don't even have a gulp coverage task for the server yet.

WDYT?

**Comment by jloveland** [comment]
also, what are your thoughts on outputting reports for client and server in `coverage/client` and `coverage/server` respectively?

**Comment by lirantal** [comment]
:+1: on adding coverage report for the client side
Sounds like a good idea to separate them for the future. People might be interested only in one aspect (client vs server) or will want to do different things with them in CI/CD process.

**Comment by ilanbiala** [comment]
:+1:

**Comment by jloveland** [comment]
@lirantal and @ilanbiala fixed conflicts, this should be ready to merge.

**Comment by ilanbiala** [comment]
@lirantal should we see coverage in Coveralls already, or will we have to merge the PR?

**Comment by lirantal** [comment]
IIRC you must merge first because it's just tracking the `master` branch.
